power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -709,25 +720,30 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         };
     }
 
-    static auto scenario_update_restore(MainModelImpl& model, ConstDataset const& update_data,
-                                        SequenceIdx const& scenario_sequence, SequenceIdx& scenario_sequence_cache,
-                                        bool is_independent, std::vector<CalculationInfo>& infos) noexcept {
-        bool const do_update_cache = !is_independent;
+    static auto scenario_update_restore(
+        MainModelImpl& model, ConstDataset const& update_data,
+        std::array<std::reference_wrapper<std::vector<Idx2D> const>, n_types> const& scenario_sequence,
+        SequenceIdx& current_scenario_sequence_cache, ComponentFlags const& is_independent,
```

Comment:
nitpick: dont change now, adding it for refactoring
```suggestion
        SequenceIdx& current_scenario_sequence_cache, ComponentFlags const& independence_flags,
``` 
Was wondering how one can tranasform a boolean haha